src/components/gridContainer.js

```diff
@@ -58,31 +63,36 @@ class GridContainer extends Component {
         this.previewPostData = this.processHandsontableData.bind(this);
         this.updateUserTableData = this.updateTableJsonOutput.bind(this);
         this.cellMove = this.cellMove.bind(this);
+        this.setDataDirty = this.setDataDirty.bind(this);
 
-        this.previewGrid = this.previewGrid.bind(this);
+        this.onPreviewGrid = this.onPreviewGrid.bind(this);
         this.rebuildGrid = this.rebuildGrid.bind(this);
         this.onBackFromPreview = this.onBackFromPreview.bind(this);
         this.saveGrid = this.saveGrid.bind(this);
         this.cancel = this.cancel.bind(this);
+        this.onError = this.onError.bind(this);
     
     }
 
 
 
     componentDidMount() {
-        console.log('preLoadData');
-        console.log(this.props.data);
-        console.log(Object.keys(this.props.data).length === 0)
-
-        if (!(Object.keys(this.props.data).length === 0))
-        {
-            this.setState({
-                handsontableData: this.props.data.data
-            })
-            this.populateMetaForm(this.props.data);
-            this.rebuildGrid(this.props.data)
-        // this.changeView('editTable');
-        }}
+        // console.log('preLoadData');
+        // console.log(this.props.data);
+       
+
+        if (this.props.data) {
+            if (!(Object.keys(this.props.data).length === 0))
```

Comment:
Is there any reason we can't do
```
if (this.props.data && !(Object.keys(this.props.data).length === 0)) {
```

instead on the line above this rather than nesting if statements?